libs/blocks/quiz/utils.js

```diff
@@ -9,9 +9,13 @@ const RESULTS_EP_NAME = 'results.json';
 let configPath; let quizKey; let analyticsType; let analyticsQuiz; let metaData;
 const { locale } = getConfig();
 
-const initConfigPath = (roolElm) => {
-  const link = roolElm.querySelector('.quiz > div > div > a');
-  const quizConfigPath = link?.text.toLowerCase();
+export const initConfigPath = (quizMetaData) => {
```

Comment:
I got enhance code from AI like this, do we really need the validation here or we are good?
/**
 * Initializes the configuration path based on the given quizMetaData.
 *
 * @param {Object} quizMetaData - The metadata of the quiz.
 * @returns {Function} The function to get the path of the file.
 */
export const initConfigPath = (quizMetaData) => {
  if (!quizMetaData || !quizMetaData.quizurl || typeof quizMetaData.quizurl.text !== 'string') {
    throw new Error('Invalid quizMetaData. Make sure it has a "quizurl.text" property of type string.');
  }

  const quizConfigPath = quizMetaData.quizurl.text.toLowerCase();
  const urlParams = new URLSearchParams(window.location.search);
  const stringsPath = urlParams.get('quiz-data');

  return (filepath) => {
    if (!filepath || typeof filepath !== 'string') {
      throw new Error('Invalid filepath. It should be a non-empty string.');
    }

    const baseConfigPath = stringsPath || quizConfigPath;
    return `${baseConfigPath}${filepath}`;
  };
};

Reply:
and When you use quizMetaData, it is the same result as before, right?

Reply:
I close this after sync with Erich, because we will have a story for the validation part